### ISSUE TITLE
java11/java17 do not publish a manifest like java11-debian11/java17-debian11, preventing architecture specific pulls

### ISSUE THREAD

**Issue by andrewcanby-finocomp** [GoogleContainerTools/distroless#1216]
Thanks for your efforts folks. I wanted to raise this for your consideration/thoughts.

**Describe the bug**
According to the `README.md`:
> Distroless images are based on Debian 11 (bullseye). Images are explicitly tagged with Debian version suffixes (e.g. -debian11). Specifying an image without the distribution will currently select -debian11 images...

My interpretation of this is `gcr.io/distroless/java11:nonroot` and `gcr.io/distroless/java11-debian11:nonroot` should currently resolve to the same image (manifest?).

This is true to a point, though seems to have some limitations for multi arch manifests. The non suffixed java versions (e.g. `java11:nonroot`) seem to only point to the `amd64` images.

This means that `docker pull gcr.io/distroless/java11:nonroot` will always pull an `amd64` distro image, irrespective of arch; but `docker pull gcr.io/distroless/java11-debian11:nonroot` can pull an `arm64` version when appropriate.

**To Reproduce**
```bash
# on a non amd64 machine, e.g. Apple M1
❯ docker pull gcr.io/distroless/java11:nonroot
❯ docker inspect gcr.io/distroless/java11:nonroot
[
    {
        "Id": "sha256:7a999fe7fc26d091cd8f1190f40a0bc981f68b07ceab3936dbe06d6e3928fc56",
        "RepoTags": [
            "gcr.io/distroless/java11:nonroot"
        ],
        "RepoDigests": [
            "gcr.io/distroless/java11@sha256:00733d19f8d511cfead7b728f730cc50a095011e3c8b68d9a7118899765824bc"
        ],
        "Architecture": "amd64",
        "Os": "linux",
        # snip
❯ docker pull gcr.io/distroless/java11-debian11:nonroot
❯ docker inspect gcr.io/distroless/java11-debian11:nonroot
[
    {
        "Id": "sha256:edb60bd68a7dcfb6c81105ff51f3b8caf07c86108d0d881615d9502ba05781e3",
        "RepoTags": [
            "gcr.io/distroless/java11-debian11:nonroot"
        ],
        "RepoDigests": [
            "gcr.io/distroless/java11-debian11@sha256:1a6b5a77e1a70991df81b7e524372ece1c01a98feebe83f351781ba8eaa0d6a0"
        ],
        "Architecture": "arm64",
        "Os": "linux",
        # snip
```

**Expected behavior**
Ideally a `docker pull` should resolve the same image, considering architecture. Should the non Debian suffixed images point to the manifest of the Debian image?

**Console Output**
N/A

**Additional context**
Some links (I find the GCR browser takes a bit to correlate things)
[`java11:nonroot`](https://console.cloud.google.com/gcr/images/distroless/global/java11@sha256:00733d19f8d511cfead7b728f730cc50a095011e3c8b68d9a7118899765824bc/details?tab=info) points at `sha256:00733d19f8d511cfead7b728f730cc50a095011e3c8b68d9a7118899765824bc` which is the [`java11-debian11:nonroot-amd64` tag](https://console.cloud.google.com/gcr/images/distroless/global/java11-debian11@sha256:00733d19f8d511cfead7b728f730cc50a095011e3c8b68d9a7118899765824bc/details?tab=info)

To be clear I can/will use the `debian11` images going forward to avoid this pitfall. Thanks again!

**Comment by andrewcanby-finocomp** [comment]
I noticed https://github.com/GoogleContainerTools/distroless#what-images-are-available mentions only the `-debian11` variants are supported, so the approach to take is obvious.

However it also mentions

> Any other tags are considered deprecated and are no longer updated

But the non distro tags are still updated: https://console.cloud.google.com/gcr/images/distroless/global/java11, so perhaps the confusion will linger for a while?

**Comment by loosebazooka** [comment]
Yeah historically, only the `<image>-<distro>` images have been mulit-arch, I think you make a good point that they should all be multi arch. We'll look into that, but we're in the middle of some major changes to the build so it might take a little bit to get to.

**Comment by andrewcanby-finocomp** [comment]
Thanks @loosebazooka - will be using `<image>-<distro>` as suggested so low priority is fine by me

**Comment by mnacken** [comment]
Any news on this?

We are using renovatebot for updating our own internal images which are based on the distroless images.

So far we have been able to keep everything up to date using the tag for example "nodejs18-debian11:nonroot-amd64@sha256:733debf656767452adbedc10c2100e988ccac7c63589c13d6a9587a2bbb74501" and the sha-digest.
Due to the removal of the "amd64" tag from the "<image>-<distro>" like images, this is no longer possible.

Is it planned that <image>-<distro> images will again be offered with the architecture tag ?

**Comment by loosebazooka** [comment]
this should be fixed with #1987
